### PR TITLE
Add `read_audio_info`, fix improper MP3 metadata read on CommonVoice/DVoice

### LINKED ISSUE #1925: [Bug]: CommonVoice sorting is likely broken with certain torchaudio versions
### Describe the bug

CommonVoice is a dataset in MP3 format. It seems that a recent `torchaudio` version (linked issue reports `0.12`, but my `0.11` environment seems to reproduce) more or less intentionally broke querying the length of MP3 files when using `torchaudio.info(path).num_frames`, as it seems to always return 0, see <https://github.com/pytorch/audio/issues/2524>.

It seems that a PR was merged at some point to solve that issue but the minimum torchaudio version we require is probably still affected.

SpeechBrain uses this to generated the CSVs. As a result, they contain a `0.0` duration column.

`train.py` uses that column to sort data.

### Expected behaviour

`duration` should never be zero.

### To Reproduce

Run any preprocessing step in a CommonVoice recipe, then inspect `results/<seed>/save/*.csv`.

### Versions

_No response_

### Relevant log output

Example contents of CSV:

```
ID,duration,wav,spk_id,wrd
common_voice_fr_19642221,0.0,/users/sdelangen/CommonVoice8.0/fr/clips/common_voice_fr_19642221.mp3,4e64587f8dd935d70e50607fe60f73b5badcd6e967db2c331a6c92ca00e9d1ddde6ff2385cd1a90c6d5ceea960e708248914dcb6cdc69fe6ea7edd0208a3ff5e,C EST LE PLUS VASTE ET LE PLUS PEUPLÉ DE LA COMMUNE
common_voice_fr_19642222,0.0,/users/sdelangen/CommonVoice8.0/fr/clips/common_voice_fr_19642222.mp3,4e64587f8dd935d70e50607fe60f73b5badcd6e967db2c331a6c92ca00e9d1ddde6ff2385cd1a90c6d5ceea960e708248914dcb6cdc69fe6ea7edd0208a3ff5e,CES SEPT QUATUORS ONT ÉTÉ ÉDITÉS PAR MARC LOMBARD
common_voice_fr_19642223,0.0,/users/sdelangen/CommonVoice8.0/fr/clips/common_voice_fr_19642223.mp3,
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AsuMagic**
Fixes #1925.

This PR introduces a `dataio` function `read_audio_info` which, for the most part, behaves identically to `torchaudio.info`. However, because certain `torchaudio` versions have a "bug" which makes it so that `num_frames` is not populated with MP3 (which was solved in very recent versions), this function additionally decodes the file to obtain the frame count.  
This is slow, but this is what newer torchaudio does anyway(? might be ffmpeg now), as there doesn't seem to be a perfectly consistent way to achieve this with MP3.

Data preparation was tested over for CommonVoice, but not for DVoice, which also uses MP3. However, because these steps are very similar, there should be no reason for it to fail.

A test was also added to verify whether it appears to function properly (and also does so for mp3+ogg+wav for good measure).

Recent versions of torchaudio offload MP3 handling to ffmpeg, so this PR also adds a ffmpeg install step for the CI (it does not touch requirements files and such, as it is often system provided). This could be split to a different PR if required.

### CODE REVIEW

**Comment by AsuMagic** [comment]
The CI error seems to hint that ffmpeg is not available for the `torchaudio` version in use, judging from where this error originates from: <https://pytorch.org/audio/stable/_modules/torchaudio/backend/sox_io_backend.html>.

I will look into it and add fix CI within the PR if I'm correct.

**Comment by Adel-Moumen** [comment]
Hi @AsuMagic,

Many thanks for this PR! 

@Gastron could you please have a look? Thanks!

**Comment by Gastron** [comment]
I can take a look, but as I understand, let's work on #1944 first, and then this.

**Review by Gastron — APPROVED** [review]
Ok now that #1944 was merged I looked at these changes. The read_audio_info is great, let's merge this after the merge conflicts (from #1944) have been resolved.

**Comment by AsuMagic** [comment]
Rebased against `develop`, contents are identical other than `read_audio_info` being moved around in CV prepare. I tested it again, it still works.

**Pull request merged**
